sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -350,6 +358,16 @@ class SessionCatalog(
     }
   }
 
+  private def makeQualifiedTablePath(locationUri: URI, database: String): URI = {
+    if (locationUri.isAbsolute) {
+      locationUri
+    } else {
+      val dbName = formatDatabaseName(database)
+      val dbLocation = makeQualifiedDBPath(getDatabaseMetadata(dbName).locationUri)
```

Comment:
I'm a bit concerned about it as it adds an extra database lookup. Is it better to push this work to the underlying external catalog?

Reply:
Do you mean that we are calling `requireDbExists(dbName)` repeatedly?

Reply:
Ideally `createTable` should only do one RPC to the hive metastore. `requireDbExists` is one problem but we can simply remove it. However, the new database lookup seems can't be easily removed.

Reply:
> However, the new database lookup seems can't be easily removed.

Yes, it seem make no difference even we put this into external catalogs, we have to call the `getDatabase` API in order to get the actual path of the database for this particular case.

Reply:
what happens if we don't qualify the path here and leave it to hive metastore? Will it still be a relative path in the hive metastore?

Reply:
FYI, https://github.com/apache/spark/pull/17254

Reply:
ok we did the same thing for partition. LGTM then